spynnaker/pyNN/models/neuron/synapse_dynamics/synapse_dynamics_structural.py

```diff
@@ -0,0 +1,567 @@
+from six import itervalues
+import numpy as np
+import collections
+
+from spinn_utilities.overrides import overrides
+from data_specification.enums.data_type import DataType
+from spynnaker.pyNN.models.neural_projections import ProjectionApplicationEdge
+from spynnaker.pyNN.models.neural_projections import ProjectionMachineEdge
+from .abstract_plastic_synapse_dynamics import AbstractPlasticSynapseDynamics
+from .abstract_synapse_dynamics_structural import \
+    AbstractSynapseDynamicsStructural
+from .synapse_dynamics_stdp import SynapseDynamicsSTDP
+from .synapse_dynamics_static import SynapseDynamicsStatic
+from spynnaker.pyNN.utilities import constants
+
+
+class SynapseDynamicsStructural(AbstractSynapseDynamicsStructural):
+    __slots__ = [
+        "_f_rew",
+        "_p_rew",
+        "_weight",
+        "_delay",
+        "_s_max",
+        "_lateral_inhibition",
+        "_sigma_form_forward",
+        "_sigma_form_lateral",
+        "_p_form_forward",
+        "_p_form_lateral",
+        "_p_elim_dep",
+        "_p_elim_pot",
+        "_grid",
+        "_random_partner",
+        "_connections",
+        "_actual_row_max_length"]
+
+    def __init__(self, stdp_model=None, f_rew=10 ** 4, weight=0, delay=1,
```

Comment:
as this is used multiple places. make these default values in a array up top, and have the other places call from that for default values. look at how models do ti for their components.